src/tests/audio/data_structures/audio_test.py

```diff
@@ -17,19 +17,124 @@
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_offset() -> None:
+    """Tests loading audio with a positive offset."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=1.0)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
+    assert isinstance(audio.sampling_rate, int)
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_offset_exceeding_duration() -> None:
+    """Tests that an excessive offset raises an error."""
+    info = torchaudio.info(MONO_AUDIO_PATH)
+    file_duration = info.num_frames / info.sample_rate
+
+    with pytest.raises(ValueError, match="Offset .* exceeds the duration of the audio file"):
+        Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=file_duration + 1.0)
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_duration() -> None:
+    """Tests loading a specific duration of an audio file."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, duration_in_sec=2.0)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
+    assert isinstance(audio.sampling_rate, int)
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_with_offset_and_duration() -> None:
+    """Tests loading audio with both an offset and a duration."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=1.0, duration_in_sec=2.0)
+    assert audio is not None
+    assert audio.waveform is not None
+    assert audio.waveform.shape[1] > 0
+    assert isinstance(audio.sampling_rate, int)
+    assert audio.sampling_rate == 48000
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_negative_offset() -> None:
+    """Tests that a negative offset raises an error."""
+    with pytest.raises(ValueError, match="Offset must be a non-negative value"):
+        Audio.from_filepath(MONO_AUDIO_PATH, offset_in_sec=-1.0)
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_negative_duration() -> None:
+    """Tests that a negative duration (except -1) raises an error."""
+    with pytest.raises(ValueError, match="Duration must be -1 .* or a positive value"):
+        Audio.from_filepath(MONO_AUDIO_PATH, duration_in_sec=-2.0)
+
+
+@pytest.mark.skipif(not TORCHAUDIO_AVAILABLE, reason="torchaudio is not installed.")
+def test_audio_creation_full_duration() -> None:
+    """Tests loading the full audio file with duration=-1."""
+    audio = Audio.from_filepath(MONO_AUDIO_PATH, duration_in_sec=-1)
```

Comment:
If -1 is the default then this should already be handled but better to over test than under test